publisher/pubsub.go

```diff
@@ -3,30 +3,43 @@ package publisher
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/goccy/go-json"
 )
 
 // GooglePubSubPublisher represent Pub/Sub publisher.
 type GooglePubSubPublisher struct {
 	pubSubConnection *PubSubConnection
+	enableOrdering   bool
 }
 
 // NewGooglePubSubPublisher create new instance of GooglePubSubPublisher.
-func NewGooglePubSubPublisher(pubSubConnection *PubSubConnection) *GooglePubSubPublisher {
+func NewGooglePubSubPublisher(pubSubConnection *PubSubConnection, enableOrdering bool) *GooglePubSubPublisher {
 	return &GooglePubSubPublisher{
-		pubSubConnection,
+		pubSubConnection: pubSubConnection,
+		enableOrdering:   enableOrdering,
 	}
 }
 
 // Publish send events, implements eventPublisher.
-func (p *GooglePubSubPublisher) Publish(ctx context.Context, topic string, event *Event) error {
+func (p *GooglePubSubPublisher) Publish(ctx context.Context, topic string, event *Event) PublishResult {
 	body, err := json.Marshal(event)
 	if err != nil {
-		return fmt.Errorf("marshal: %w", err)
+		return NewErrorResult(fmt.Errorf("marshal: %w", err))
+	}
+	var orderingKey string
+	if p.enableOrdering {
+		if len(event.PrimaryKey) > 0 {
```

Comment:
If we have ordering enabled but no primary key this table will be limited to 1Mbps of throughput?

Should we warn or error out in that case?